crates/topos-api/src/graphql/checkpoint.rs

```diff
@@ -0,0 +1,43 @@
+use async_graphql::InputObject;
+use serde::{Deserialize, Serialize};
+
+use topos_uci::{self, SUBNET_ID_LENGTH};
+
+#[derive(Debug, Default, Serialize, Deserialize, InputObject)]
+pub struct CertificateId {
+    pub value: String,
+}
+
+#[derive(Clone, Debug, Default, Serialize, Deserialize, InputObject)]
+pub struct InputSubnetId {
+    pub value: String,
+}
+
+#[derive(Debug, Default, Serialize, Deserialize, InputObject)]
+pub struct SourceStreamPosition {
+    pub source_subnet_id: InputSubnetId,
+    pub position: u64,
+    pub certificate_id: Option<CertificateId>,
+}
+
+#[derive(Debug, Default, Serialize, Deserialize, InputObject)]
+pub struct SourceCheckpoint {
+    pub source_subnet_ids: Vec<InputSubnetId>,
+    pub positions: Vec<SourceStreamPosition>,
+}
+
+impl InputSubnetId {
+    pub fn to_uci_subnet_id(self) -> topos_uci::SubnetId {
```

Comment:
nit: implementing `into`? or is it to reflect a potential costly conversion?

Reply:
The problem is that `topos_uci::SubnetId` is in a different module. We would have to do the `From<SubnetId` in the module it is defined, which creates a cyclic dependency. It's not the nicest solution, but I didn't want to do such a big refactor to restructure the modules.

Reply:
@gruberb It is possible as we already do that for the `gRPC` ones. I will create a PR to fix that :)

Reply:
You mean importing the `topos_uci::SubnetId` somewhere else? I didn't want to touch the `gRPC` implementation for a `GraphQL` use case. Otherwise, to `impl` something on a type, AFAIK, you cannot import it! But curious what you mean exactly :)